62.CAD/vertex_shader.hlsl

```diff
@@ -398,20 +445,71 @@ PSInput main(uint vertexID : SV_VertexID)
         outV.position.xy = (outV.position.xy / globals.resolution) * 2.0 - 1.0;
 
     }
-    /*
-        TODO[Lucas]:
-        Another `else if` for CurveBox Object Type,
-        What you basically need to do here is transform the box min,max and set `outV.position` correctly based on that + vertexIdx
-        and you need to do certain outV.setXXX() functions to set the correct (transformed) data to frag shader
-
-        Another note: you may know that for transparency we draw objects twice
-        only when `writeToAlpha` is true (even provoking vertex), sdf calculations will happen and alpha will be set
-        otherwise it's just a second draw to "Resolve" and the only important thing on "Resolves" is the same `outV.position` as the previous draw (basically the same cage)
-        So if you do any precomputation, etc for sdf caluclations you could skip that :D and save yourself the trouble if `writeToAlpha` is false.
-    */
-    
-    
-// Make the cage fullscreen for testing:
+    else if (objType == ObjectType::CURVE_BOX)
+    {
+        outV.setColor(lineStyle.color);
+        outV.setLineThickness(screenSpaceLineWidth / 2.0f);
+
+        CurveBox curveBox;
+        curveBox.aabbMin = vk::RawBufferLoad<double2>(drawObj.geometryAddress, 8u);
+        curveBox.aabbMax = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2), 8u);
+        curveBox.curveMin[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 2, 8u);
+        curveBox.curveMin[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 3, 8u);
+        curveBox.curveMin[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 4, 8u);
+        curveBox.curveMax[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 5, 8u);
+        curveBox.curveMax[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 6, 8u);
+        curveBox.curveMax[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 7, 8u);
+
+        const bool2 maxCorner = bool2(vertexIdx & 0x1u, vertexIdx >> 1);
+        const double2 coord = transformPointNdc(lerp(curveBox.aabbMin, curveBox.aabbMax, maxCorner));
+        outV.position = float4((float2) coord, 0.f, 1.f);
+
+        const uint major = 1;
+        const uint minor = 1-major;
+
+        nbl::hlsl::shapes::QuadraticBezier curveMin = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMin[0], curveBox.curveMin[1], curveBox.curveMin[2]);
+        nbl::hlsl::shapes::QuadraticBezier curveMax = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMax[0], curveBox.curveMax[1], curveBox.curveMax[2]);
+
```

Comment:
How you work with and evaluate beziers will change after I merge @Przemog1's PR
To get your mind ready about the changes take a look at this PR and how we evaluate beziers:

https://github.com/Devsh-Graphics-Programming/Nabla/pull/527/files#diff-aeb209a11841655ba7cebc9d61b48153770cfa0c95c6d8367ab58c29e01a61db

https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/53/files#diff-f1b9d90ab290c51d50a8e604e67bfd4dbc156ec2b1209d420313780d368069d9